src/data_source/storage/sql/queries/explorer.rs

```diff
@@ -105,6 +105,164 @@ where
     }
 }
 
+lazy_static::lazy_static! {
+    static ref GET_BLOCK_SUMMARIES_QUERY_FOR_LATEST: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                ORDER BY h.height DESC
+                LIMIT $1"
+            )
+    };
+
+    static ref GET_BLOCK_SUMMARIES_QUERY_FOR_HEIGHT: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                WHERE h.height <= $1
+                ORDER BY h.height DESC
+                LIMIT $2"
+        )
+    };
+
+    // We want to match the blocks starting with the given hash, and working backwards
+    // until we have returned up to the number of requested blocks.  The hash for a
+    // block should be unique, so we should just need to start with identifying the
+    // block height with the given hash, and return all blocks with a height less than
+    // or equal to that height, up to the number of requested blocks.
+    static ref GET_BLOCK_SUMMARIES_QUERY_FOR_HASH: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                WHERE h.height <= (SELECT h1.height FROM header AS h1 WHERE h1.hash = $1)
+                ORDER BY h.height DESC
+                LIMIT $2",
+        )
+    };
+
+    static ref GET_BLOCK_DETAIL_QUERY_FOR_LATEST: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                ORDER BY h.height DESC
+                LIMIT 1"
+        )
+    };
+
+    static ref GET_BLOCK_DETAIL_QUERY_FOR_HEIGHT: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                WHERE h.height = $1
+                ORDER BY h.height DESC
+                LIMIT 1"
+        )
+    };
+
+    static ref GET_BLOCK_DETAIL_QUERY_FOR_HASH: String = {
+        format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                WHERE h.hash = $1
+                ORDER BY h.height DESC
```

Comment:
same here. 
 The block hash is unique so I don't understand the purpose of ORDER BY here. Is this because of forcing the query planner to use index search?